modules/kafka/src/main/java/org/apache/ignite/stream/kafka/connect/IgniteSourceTask.java

```diff
@@ -0,0 +1,309 @@
+
+    /** Event buffer size. */
+    private static int evtBufSize = 100000;
+
+    /** Event buffer. */
+    private static BlockingQueue<CacheEvent> evtBuf = new LinkedBlockingQueue<>(evtBufSize);
+
+    /** Max number of events taken from the buffer at once. */
+    private static int evtBatchSize = 100;
+
+    /** Flag for stopped state. */
+    private static volatile boolean stopped = true;
+
+    /** Ignite grid configuration file. */
+    private static String igniteConfigFile;
+
+    /** Cache name. */
+    private static String cacheName;
+
+    /** Listener id. */
+    private static UUID rmtLsnrId;
+
+    /** User-defined filter. */
+    private static IgnitePredicate<CacheEvent> filter;
+
+    /** Topic. */
+    private static String topics[];
+
+    /** Offset. */
+    private static final Map<String, Long> offset = Collections.singletonMap("offset", 0L);
+
+    /** Partition. */
+    private static final Map<String, String> srcPartition = Collections.singletonMap("cache", null);
+
+    /** {@inheritDoc} */
+    @Override public String version() {
+        return new IgniteSinkConnector().version();
+    }
+
+    /**
+     * Filtering is done remotely. Local listener buffers data for injection into Kafka.
+     *
+     * @param props Task properties.
+     */
+    @Override public void start(Map<String, String> props) {
+        // Each task has the same parameters -- avoid setting more than once.
+        if (cacheName != null)
+            return;
+
+        cacheName = props.get(IgniteSourceConstants.CACHE_NAME);
+        igniteConfigFile = props.get(IgniteSourceConstants.CACHE_CFG_PATH);
+        topics = props.get(IgniteSourceConstants.TOPIC_NAMES).split("\\s*,\\s*");
+
+        if (props.containsKey(IgniteSourceConstants.INTL_BUF_SIZE))
+            evtBufSize = Integer.parseInt(props.get(IgniteSourceConstants.INTL_BUF_SIZE));
+
+        if (props.containsKey(IgniteSourceConstants.INTL_BATCH_SIZE))
+            evtBatchSize = Integer.parseInt(props.get(IgniteSourceConstants.INTL_BATCH_SIZE));
+
+        if (props.containsKey(IgniteSourceConstants.CACHE_FILTER_CLASS)) {
+            String filterCls = props.get(IgniteSourceConstants.CACHE_FILTER_CLASS);
+            if (filterCls != null && !filterCls.isEmpty()) {
+                try {
+                    Class<? extends IgnitePredicate<CacheEvent>> clazz =
+                        (Class<? extends IgnitePredicate<CacheEvent>>)Class.forName(filterCls);
+
+                    filter = clazz.newInstance();
+                }
+                catch (Exception e) {
+                    log.error("Failed to instantiate the provided filter! " +
+                        "User-enabled filtering is ignored!", e);
+                }
+            }
+        }
+
+        IgniteBiPredicate<UUID, CacheEvent> locLsnr = new IgniteBiPredicate<UUID, CacheEvent>() {
+            @Override public boolean apply(UUID id, CacheEvent evt) {
```

Comment:
Please don't use anonymous classes for both remote and local listener because in that case their enclosing class (IgniteSourceTask) will be serialized and sent over the wire along with them. Just move implementation of the listeners to "private static class".